lib/route.coffee

```diff
@@ -207,8 +207,21 @@ class @Route
     # Send response
     endpointContext.response.writeHead statusCode, headers
     endpointContext.response.write body
-    endpointContext.response.end()
-
+    if statusCode in [401, 403]
+      # This is a simple, but not really adequate way to slow down unauthorized scans of the server.
+      # It is not sufficient because hackers can DDOS across multiple threads and systems.
+      # See https://www.owasp.org/index.php/Blocking_Brute_Force_Attacks#Finding_Other_Countermeasures
+      delayInMilliseconds = 500
+      # Hackers can measure the response time to determine things like whether the 401 response was 
+      # caused by bad user id vs bad password.
+      # In doing so, they can first scan for valid user ids regardless of valid passwords.
+      # Delay by a random amount to reduce the ability for a hacker to determine the response time.
+      # See https://en.wikipedia.org/wiki/Timing_attack
```

Comment:
Updated variable names. Added more comments to help understand why we do this.
Also added some links.